core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartUpscaleCommissionViaCMHandler.java

```diff
@@ -96,7 +96,7 @@ protected Selectable doAccept(HandlerEvent<StopStartUpscaleCommissionViaCMReques
                     String.valueOf(allInstancesToCommission.size()));
 
             ClusterSetupService clusterSetupService = clusterApiConnectors.getConnector(stack).clusterSetupService();
-            clusterSetupService.waitForHostsHealthy(new HashSet<>(allInstancesToCommission));
+            allInstancesToCommission = clusterSetupService.waitForHostsHealthy(new HashSet<>(allInstancesToCommission));
 
             flowMessageService.fireEventAndLog(stack.getId(), UPDATE_IN_PROGRESS.name(), CLUSTER_SCALING_STOPSTART_UPSCALE_CMHOSTSSTARTED,
```

Comment:
We should also log failed hosts so that those can be monitored. If few hosts took more than configured timeout to come up, they will be treated as failed i.e. decommissioned in CM but they will be actually be running on cloud provider.